client/my-sites/plugins/plugin-custom-domain-dialog/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { Dialog } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import { getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import PluginIcon from '../plugin-icon/plugin-icon';
+import './style.scss';
+
+interface Props {
+	plugin: { name: string; icon: string };
+	domains: Array< { isPrimary: boolean; isSubdomain: boolean } >;
+	closeDialog: () => void;
+	isDialogVisible: boolean;
+	onProceed: () => void;
+}
+
+export const PluginCustomDomainDialog = ( {
+	plugin,
+	domains,
+	closeDialog,
+	isDialogVisible,
+	onProceed,
+}: Props ): JSX.Element => {
+	const translate = useTranslate();
+	const selectedSiteUrl = useSelector( ( state ) => getSelectedSiteSlug( state ) );
+
+	const hasNonPrimaryCustomDomain = domains.some(
+		( { isPrimary, isSubdomain } ) => ! isPrimary && ! isSubdomain
+	);
+
+	const buttons = [
+		{
+			action: 'learn-more',
+			href: '/support/domains/#set-a-primary-address',
```

Comment:
We should use a localised absolute link here

```suggestion
			href: localizeUrl( 'https://wordpress.com/support/domains/#set-a-primary-address' ),
```